Mistica/Source/Components/EmptyState/EmptyState.swift

```diff
@@ -0,0 +1,134 @@
+                configure(withConfiguration: contentConfiguration)
+            } else {
+                configure(withConfiguration: .emptyConfiguration)
+            }
+        }
+    }
+
+    override public var accessibilityElements: [Any]? {
+        get {
+            // We must set the frame and be sure it is already calculated.
+            emptyStateAccessibilityElement.accessibilityFrameInContainerSpace = bounds
+            return [
+                emptyStateAccessibilityElement,
+                emptyStateContentBase
+            ].compactMap { $0 }
+        }
+        set {}
+    }
+
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        commomInit()
+    }
+
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commomInit()
+    }
+
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+    }
+
+    override public var intrinsicContentSize: CGSize {
+        emptyStateContentBase.intrinsicContentSize
+    }
+
+    override public func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        guard traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle else { return }
+        addBorder(color: .border)
+    }
+}
+
+// MARK: Public
+
+public extension EmptyState {
+    var iconContentMode: UIView.ContentMode {
+        get {
+            emptyStateContentBase.iconContentMode
+        }
+        set {
+            emptyStateContentBase.iconContentMode = newValue
+        }
+    }
+
+    var primaryButton: Button {
+        emptyStateContentBase.emptyStateActions.primaryButton
+    }
+
+    var secondaryButton: Button {
+        emptyStateContentBase.emptyStateActions.secondaryButton
+    }
+
+    var linkButton: Button {
+        emptyStateContentBase.emptyStateActions.linkButton
+    }
+
+    override var accessibilityTraits: UIAccessibilityTraits {
+        get {
+            emptyStateAccessibilityElement.accessibilityTraits
+        }
+        set {
+            emptyStateAccessibilityElement.accessibilityTraits = newValue
+        }
+    }
+}
+
+// MARK: Private
+
+private extension EmptyState {
+    func commomInit() {
+        layoutViews()
+        styleViews()
+    }
+
+    func layoutViews() {
+        directionalLayoutMargins = Constants.emptyStateLayoutMargins
+        insetsLayoutMarginsFromSafeArea = false
+    }
+
+    func styleViews() {
+        backgroundColor = .background
+    }
+
+    func configure(withConfiguration configuration: EmptyStateConfiguration) {
+        if configuration.isCard {
+            addSubview(constrainedToLayoutMarginsGuideOf: emptyStateContentBase)
+            addBorder(color: .border)
+            makeRounded(cornerRadius: 4.0)
+        } else {
+            addSubview(withDefaultConstraints: emptyStateContentBase)
+        }
```

Comment:
This method can be called more than once. We may need to remove the border and the corner radius, and make sure we remove emptyEstateContentBase before adding it again a subview.
```suggestion
        emptyStateContentBase.removeFromSuperview()
        if configuration.isCard {
            addSubview(constrainedToLayoutMarginsGuideOf: emptyStateContentBase)
            addBorder(color: .border)
            makeRounded(cornerRadius: 4.0)
        } else {
            addSubview(withDefaultConstraints: emptyStateContentBase)
            removeBorder()
            makeRounded(cornerRadius: 0)
        }
```